python/BioSimSpace/Align/_align.py

```diff
@@ -1297,13 +1763,140 @@ def flexAlign(
     return _Molecule(molecule0)
 
 
+def _roiAlign(
+    molecule0,
+    molecule1,
+    roi,
+    align_function,
+    property_map0,
+    property_map1,
+    fkcombu_exe=None,
+):
+    """
+    Flexibly align residue of interest (ROI) in molecule0 to that in molecule1
+    using BioSimSpace.Align._flexAlign().
+
+    Parameters
+    ----------
+
+    molecule0 : :class:`Molecule <BioSimSpace._SireWrappers.Molecule>`
+        The molecule to align.
+
+    molecule1 : :class:`Molecule <BioSimSpace._SireWrappers.Molecule>`
+        The reference molecule.
+
+    roi : list
+        The region of interest to align.
+        Consists of a list of ROI residue indices.
+
+    align_function : str
+        The alignment function used to align atoms. Available options are:
+            - "rmsd"
+                Align atoms in molecule0 to those in molecule1 using the mapping
+                between matched atom indices.
+                Uses :class:`rmsdAlign <BioSimSpace.Align._rmsdAlign>` to align the atoms in the ROI.
+            - "rmsd_flex_align"
+                Flexibly align roi from molecule0 to molecule1 based on the mapping.
+                Uses :class:`flexAlign <BioSimSpace.Align._flexAlign>` to align the atoms in the ROI.
+
+    fkcombu_exe : str
+        Path to the fkcombu executable. Will only be used if aligning with
+        "rmsd_flex_align" function. If None is passed, then BioSimSpace
+        will attempt to find fkcombu by searching your PATH.
+
+    property_map0 : dict
+        A dictionary that maps "properties" in molecule0 to their user
+        defined values. This allows the user to refer to properties
+        with their own naming scheme, e.g. { "charge" : "my-charge" }
+
+    property_map1 : dict
+        A dictionary that maps "properties" in molecule1 to their user
+        defined values.
+
+    Returns
+    -------
+
+    molecule : :class:`Molecule <BioSimSpace._SireWrappers.Molecule>`
+        The aligned molecule.
+    """
+
+    if not isinstance(molecule0, _Molecule):
+        raise TypeError(
+            "'molecule0' must be of type 'BioSimSpace._SireWrappers.Molecule'"
+        )
+
+    if not isinstance(molecule1, _Molecule):
+        raise TypeError(
+            "'molecule1' must be of type 'BioSimSpace._SireWrappers.Molecule'"
+        )
+    if roi is None:
+        raise ValueError("No region of interest (ROI) has been provided.")
+    else:
+        if not isinstance(roi, list):
+            raise TypeError("'roi' must be of type 'list'.")
```

Comment:
Extra validation of `roi` as described in previous comment.

Reply:
Same as in the previous comment.